cylc/flow/config_schema.py

```diff
@@ -0,0 +1,660 @@
+            'abort on stalled': [VDR.V_BOOLEAN],
+            'abort on timeout': [VDR.V_BOOLEAN],
+            'abort on inactivity': [VDR.V_BOOLEAN],
+            'mail events': [VDR.V_STRING_LIST, None],
+            'mail from': [VDR.V_STRING],
+            'mail smtp': [VDR.V_STRING],
+            'mail to': [VDR.V_STRING],
+            'mail footer': [VDR.V_STRING],
+        },
+        'reference test': {
+            'suite shutdown event handler': [
+                VDR.V_STRING, 'cylc hook check-triggering'],
+            'required run mode': [
+                VDR.V_STRING,
+                '', 'live', 'simulation', 'dummy-local', 'dummy'],
+            'allow task failures': [VDR.V_BOOLEAN],
+            'expected task failures': [VDR.V_STRING_LIST],
+            'live mode suite timeout': [
+                VDR.V_INTERVAL, DurationFloat(60)],
+            'dummy mode suite timeout': [
+                VDR.V_INTERVAL, DurationFloat(60)],
+            'dummy-local mode suite timeout': [
+                VDR.V_INTERVAL, DurationFloat(60)],
+            'simulation mode suite timeout': [
+                VDR.V_INTERVAL, DurationFloat(60)],
+        },
+        'authorization': {
+            # Allow owners to grant public shutdown rights at the most, not
+            # full control.
+            'public': (
+                [VDR.V_STRING, '']
+                + [level.name.lower().replace('_', '-') for level in [
+                    Priv.IDENTITY, Priv.DESCRIPTION, Priv.STATE_TOTALS,
+                    Priv.READ, Priv.SHUTDOWN]])
+        },
+    },
+    'document viewers': {
+        'html': [VDR.V_STRING, 'firefox'],
+    },
+    'documentation': {
+        'local': [VDR.V_STRING, ''],
+        'online': [VDR.V_STRING,
+                   'http://cylc.github.io/doc/built-sphinx/index.html'],
+        'cylc homepage': [VDR.V_STRING, 'http://cylc.github.io/'],
+    },
+    'editors': {
+        'terminal': [VDR.V_STRING, 'vim'],
+        'gui': [VDR.V_STRING, 'gvim -f'],
+    },
+    'job platforms': {
+        'default platform': {
+            'run directory': [VDR.V_STRING, '$HOME/cylc-run'],
+            'work directory': [VDR.V_STRING, '$HOME/cylc-run'],
+            'task communication method': [
+                VDR.V_STRING, 'default', 'ssh', 'poll'],
+            'submission polling intervals': [VDR.V_INTERVAL_LIST],
+            'execution polling intervals': [VDR.V_INTERVAL_LIST],
+            'scp command': [
+                VDR.V_STRING, 'scp -oBatchMode=yes -oConnectTimeout=10'],
+            'ssh command': [
+                VDR.V_STRING, 'ssh -oBatchMode=yes -oConnectTimeout=10'],
+            'use login shell': [VDR.V_BOOLEAN, True],
+            'login hosts': [VDR.V_STRING, ''],
+            'batch system': [VDR.V_STRING, ''],
+            'cylc executable': [VDR.V_STRING, 'cylc'],
+            'global init-script': [VDR.V_STRING],
+            'copyable environment variables': [VDR.V_STRING_LIST],
```

Comment:
This was used to communicate Rose and FCM versions, for example. Need to figure out how to do this in the world of Conda, where hopefully, Rose will be part of that environment.

Reply:
Will we need it all all? Could it's usage not be replaced by the future `Cylc-version` or by direct python references to module `__version__` variables?

Reply:
We should no longer need this setting, but we'll need a way to match deployment environments between the workflow server and *remote* job platforms.